test/unit/dialects/mssql.js

```diff
@@ -0,0 +1,37 @@
+// global it, describe
+
+'use strict';
+const expect = require('chai').expect;
+const knex = require('../../../knex');
+
+describe('MSSQL unit tests', () => {
+  const knexInstance = knex({
+    client: 'mssql',
+    connection: {
+      port: 1433,
+      host: '127.0.0.1',
+      password: 'yourStrong(!)Password',
+      user: 'sa',
+    },
+  });
+
+  it('should escape statements with [] correctly', async () => {
+    const sql = knexInstance('projects')
+      .where('id] = 1 UNION SELECT 1, @@version -- --', 1)
+      .toSQL();
+    console.log(sql);
+    expect(sql.sql).to.equal(
+      'select * from [projects] where "id] = 1 UNION SELECT 1, @@version -- --" = ?'
```

Comment:
I would like this to be verified with integration test + to test identifier that has both `"` and `[]` chars in it.

Reply:
(I had no idea that mssql supports `"` quoting for identifiers.

Reply:
[] are illegal symbols as per MSSQL documentation, apparently, so we don't need to test that. As advised by snyk people, I replaced our escaping logic with the one that `sequelize` uses, which removes `[]` in the first place.
And MSSQL does not support arrays.